compiler/rustc_parse/src/lib.rs

```diff
@@ -74,8 +75,36 @@ pub fn new_parser_from_file<'a>(
     sp: Option<Span>,
 ) -> Result<Parser<'a>, Vec<Diag<'a>>> {
     let source_file = psess.source_map().load_file(path).unwrap_or_else(|e| {
-        let msg = format!("couldn't read {}: {}", path.display(), e);
+        let msg = format!("couldn't read `{}`: {}", path.display(), e);
```

Comment:
the error here could be an enum of an io error or a utf8 error, and store the information about the span and message to be reported, and that would probably help with deduplicating these code